samtranslator/swagger/swagger.py

```diff
@@ -898,31 +900,38 @@ def add_resource_policy(self, resource_policy, path, api_id, stage):
             resource_list = self._get_method_path_uri_list(path, api_id, stage)
             self._add_ip_resource_policy_for_method(ip_range_blacklist, "IpAddress", resource_list)
 
-        if (
-            (source_vpc_blacklist is not None)
-            or (source_vpc_intrinsic_blacklist is not None)
-            or (source_vpce_intrinsic_blacklist is not None)
-        ):
-            blacklist_dict = {
-                "StringEndpointList": source_vpc_blacklist,
-                "IntrinsicVpcList": source_vpc_intrinsic_blacklist,
-                "IntrinsicVpceList": source_vpce_intrinsic_blacklist,
-            }
-            resource_list = self._get_method_path_uri_list(path, api_id, stage)
-            self._add_vpc_resource_policy_for_method(blacklist_dict, "StringEquals", resource_list)
+        if source_vpc_blacklist is not None and not all(isinstance(x, string_types) for x in source_vpc_blacklist):
+            raise InvalidDocumentException(
```

Comment:
We should maybe raise an `InvalidResourceException` so that we don't exit `translate` and also list the other potential errors in the template.

Reply:
This is a good callout but I was just following how the rest of the class handles the errors. `InvalidDocumentException` is thrown. My guess after looking into it further is  `InvalidResourceException` needs a LogicalId and it doesn't seem like the swagger class/file current has access. I am not sure if there are other reasons for this either, just what I currently understand just looking at things.

Let me spend a little time on this to see how good or bad this will be. My initially thinking is to just keep it as that is what we currently do for the rest of the errors in this category but could be worth trying to get this updated to the `InvalidResourceException`.

Reply:
Yes indeed, this class is not working with Logical IDs/resources.

Ideally we should throw an `InvalidResourcePolicyException` and catch it in the caller to raise the appropriate exception:
- `InvalidResourceException` in [api_generator.py](https://github.com/aws/serverless-application-model/blob/develop/samtranslator/model/api/api_generator.py#L588)
- `InvalidEventException` in [push.py](https://github.com/aws/serverless-application-model/blob/develop/samtranslator/model/eventsources/push.py#L749) and [events.py](https://github.com/aws/serverless-application-model/blob/develop/samtranslator/model/stepfunctions/events.py#L415)

Reply:
So I agree though this will lead to confusion to customers. The way SAM deals with APIs that are defined though the `Events` Property on a `AWS::Lambda::Function` is by first creating a [`AWS::Serverless::API` through a 'plugin'](https://github.com/aws/serverless-application-model/blob/develop/samtranslator/plugins/api/implicit_api_plugin.py). So by the time we hit this code path, there are a couple possible logical ids. We don't track the Logical Id(s) that produced this resource, so we would output the "default name logical id" we create. Customers do not have this resource (strictly) within their template so it will be very confusing. This could be why its an `InvalidDocumentException` (though I am just guessing as I don't really know).